apps/web/src/app/api/payment/customer-portal/subscriptions/[subscriptionId]/route.ts

```diff
@@ -0,0 +1,139 @@
+import { createPolarClient } from '@tuturuuu/payment/polar/client';
+import { createClient } from '@tuturuuu/supabase/next/server';
+import { getCurrentSupabaseUser } from '@tuturuuu/utils/user-helper';
+import { type NextRequest, NextResponse } from 'next/server';
+
+// PATCH: Reactivate subscription (cancel_at_period_end = false)
+export async function PATCH(
+  _req: NextRequest,
+  { params }: { params: Promise<{ subscriptionId: string }> }
+) {
+  const user = await getCurrentSupabaseUser();
+
+  if (!user) {
+    return NextResponse.json({ error: 'Unauthorized' }, { status: 401 });
+  }
+
+  const { subscriptionId } = await params;
+
+  if (!subscriptionId) {
+    return NextResponse.json(
+      { error: 'Subscription ID is required' },
+      { status: 400 }
+    );
+  }
+
+  try {
+    const supabase = await createClient();
+
+    const { data: subscription } = await supabase
+      .from('workspace_subscription')
+      .select('*')
+      .eq('id', subscriptionId)
+      .single();
+
+    if (!subscription) {
+      return NextResponse.json(
+        { error: 'Subscription not found' },
+        { status: 404 }
+      );
+    }
```

Comment:
Missing check for workspace membership and permission (must be consistent with subscription management above)